linked_list/main.py

```diff
@@ -42,54 +42,103 @@ class Node(object):
     """
     Node class representing each of the linked nodes in the list.
     """
-
     def __init__(self, elem, next=None):
-        pass
+        self.elem = elem
+        self.next = next
 
     def __str__(self):
-        pass
+        return str(self.elem)
 
     def __eq__(self, other):
-        pass
+        return isinstance(other, Node) and self.elem == other.elem
 
     def __repr__(self):
-        pass
+        return "Node: {0} with next node: {1}".format(self.elem, self.next)
 
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
-
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+            
+        if elements:
+            for element in elements:
+                self.append(element)
 
     def __str__(self):
-        pass
+        return '[{}]'.format(', '.join(str(i) for i in self))
 
     def __len__(self):
-        pass
+        return sum(1 for i in self)
```

Comment:
This is much simpler than what we implemented. We basically kept a running tally that was updated with every append() or pop() and returned that instead. I'll have to keep this in mind for future reference